### DISCUSSION TITLE
Unable to perform AC voltage parameter sweep

### DISCUSSION THREAD

**Discussion by loobylooby** [ra3xdh/qucs_s#724] [Q&A] [answered]
Hello, I wanted to do a parameter sweep of an AC voltage for a test circuit, but it gives an error. Do you know if my setup is incorrect?
I'm on Linux Mint using QUCS 24.2.1.

![image](https://github.com/ra3xdh/qucs_s/assets/94703309/58788841-c82e-42e9-a06e-320c22f34f9e)

Here is the error:

```
Ngspice started...
    Simulation interrupted due to error!

Note: No ".plot", ".print", or ".fourier" lines; no simulations run

No compatibility mode selected!


Circuit: * qucs 24.2.1  /home/seb/projects/qucs/test.sch

Error on line 6 :
  v1 in 0 dc 0 sin(0    1.000000000000000e+00     100 0 0 0) ac ( 1 0 )    1.000000000000000e+00     acphase 0
no such parameter on this device
Simulation finished

```

**Comment by ra3xdh** [comment]
I cannot reproduce this error. Everything works as expected on my machine using Ngspice-42. What Ngspice version are you using? The Ngspice-27 from Ubuntu-22.04 doesn't support variable sweep. 
![image](https://github.com/ra3xdh/qucs_s/assets/4920080/8443e01e-ed04-4533-8752-1b17f4730054)

**Reply by loobylooby** [reply]
I'm currently using ngspice-36. should I just update to ngspice-42?

**Comment by ra3xdh** [comment]
Yes, it's better to upgrade the Nsgpice. The versions before 37 also don't support S-parameter simulation.

**Reply by loobylooby** [reply]
I updated to ngspice-42 and this is my new output, which is not correctly outputting, which is strange.

![image](https://github.com/ra3xdh/qucs_s/assets/94703309/e125739d-bcdb-40b5-be55-d2d38c1b84e4)

**Comment by ra3xdh** [comment]
It is a known bug. Every transient waveform from Ngspice may have different length, because the time step is adjusted dynamically. Qucs-S plotting engine cannot process a set of waveforms of different length in one plot. I am unable to fix this without a redesign of the plotting engine. There is no workaround. The AC simulation is not affected.

**Reply by ivandi69** [reply]
The **INTERP** option can be used to reduce the number of points in the transient simulation dataset.
One can use **INCLUDE SCRIPT** component to place **.OPTION INTERP** in the beginning of the netlist. That way the number of points in the dataset is equal to the points calculated in the transient simulation dialog.

See Ngspice User's Manual section **15.1.4 Transient Analysis Options**

Here is an example: 
[buck_converter.sch.txt](https://github.com/ra3xdh/qucs_s/files/15444008/buck_converter.sch.txt)

**Reply by ra3xdh** [reply]
Thanks for pointing this. I will consider to add this option automatically if ParSweep+Tran presents on schematic.

**Comment by ra3xdh** [comment, marked as answer]
You may use the tuner instead of param.sweep to investigate the parametric circuits in transient domain.

**Reply by loobylooby** [reply]
That's what I'll do instead. Thank you!
